compiler/x/codegen/ControlFlowEvaluator.cpp

```diff
@@ -462,12 +462,33 @@ TR::Register *OMR::X86::TreeEvaluator::tableEvaluator(TR::Node *node, TR::CodeGe
          cg->evaluate(secondChild->getFirstChild()); // evaluate the glRegDeps
       }
 
-   TR::MemoryReference *tempMR = generateX86MemoryReference((TR::Register *)NULL,
-                                                            selectorReg,
-                                                            (uint8_t)(TR::Compiler->target.is64Bit()? 3 : 2),
-                                                            (intptrj_t)branchTable, cg);
+   TR::MemoryReference *jumpMR = NULL;
+   TR::Register *jumpTableReg = NULL;
+   if (TR::Compiler->target.is64Bit() && cg->comp()->compileRelocatableCode())
+      {
+      // Generate position-independent code so that no (external) relocation is
+      // necessary:
+      //
+      //    lea rJumpTable, [rip + OFFSET_TO_JUMP_TABLE]
+      //    jmp qword ptr [rJumpTable + 8*rIndex]
+      //
+      TR::LabelSymbol *label = generateLabelSymbol(cg);
+      label->setCodeLocation(reinterpret_cast<uint8_t*>(branchTable));
```

Comment:
Minor pedantic comment.  You seem to now be referring to the previously named "branchTable" as a "jumpTable".  I'm fine with that, but can we keep the naming consistent in this function?  Simply renaming `branchTable` to `jumpTable` is probably easiest as there are only a couple of references in this function.

Reply:
Renamed using "branch" instead of "jump" to stay consistent with not just `branchTable` but also `numBranchTableEntries` and `addMetaDataForBranchTableAddress()`